view/theme/frio/js/theme.js

```diff
@@ -764,12 +764,52 @@ function htmlToText(htmlString) {
  * @param {boolean} un    Whether to perform an activity removal instead of creation
  */
 function doActivityItemAction(ident, verb, un) {
-	if (verb.indexOf("attend") === 0) {
-		$(".item-" + ident + " .button-event:not(#" + verb + "-" + ident + ")").removeClass("active");
-	}
-	$("#" + verb + "-" + ident).toggleClass("active");
+  console.log(ident, verb, un);
+	_verb = un ? 'un' + verb : verb;
+	$('#like-rotator-' + ident.toString()).show();
+	$.post('item/' + ident.toString() + '/activity/' + _verb)
+	.success(
+		function(data){
+      console.log("data.status: " + data.status);
+      if (data.status == "ok") {
+        console.log("connection: " + data.status);
+        $('#like-rotator-' + ident.toString()).hide();
+        if (verb.indexOf("announce") === 0 ) {
+          console.log("announce")
+          if (data.verb == "un" + verb) {
+            $("button[id^=shareMenuOptions-" + ident.toString() + "]" ).removeClass("active");
+            $("button[id^=shareMenuOptions-" + ident.toString() + "]" ).attr("onclick", "doActivityItemAction(" + ident +", '" + verb + "', " + false + ")").change();
+            $("button[id^=" + verb + "-" + ident.toString() + "]" ).removeClass("active");
+            $("button[id^=" + verb + "-" + ident.toString() + "]" ).attr("onclick", "doActivityItemAction(" + ident +", '" + verb + "', " + false + ")").change();
+          } else {
+            $("button[id^=shareMenuOptions-" + ident.toString() + "]" ).addClass("active");
```

Comment:
In general, for asynchronous actions, you don't want to wait for the response to change the status of the interactive element. If there's no immediate feedback like a page change or a state change (the rotator isn't enough), the user may think the action didn't go through and they may click again.

Here's a better workflow:
1. User interacts with element.
2. Element changes state immediately.
3. Asynchronous request is performed.
4. On response, the display of the element is updated again depending on the success of the request.

This means that sometimes the display will flicker if the action wasn't successful, and it's okay on its own, we just need an additional piece of feedback that the action failed like you did in the `error` callback.